subs/pantry/src/Pantry/Storage.hs

```diff
@@ -729,6 +909,88 @@ sinkHackagePackageNames predicate sink = do
       cnt <- count [HackageCabalName ==. nameid]
       pure $ cnt > 0
 
+-- | Get the filename for the cabal file in the given directory.
+--
+-- If no .cabal file is present, or more than one is present, an exception is
+-- thrown via 'throwM'.
+--
+-- If the directory contains a file named package.yaml, hpack is used to
+-- generate a .cabal file from it.
+findOrGenerateCabalFile
+    :: forall env. (HasPantryConfig env, HasLogFunc env, HasProcessContext env)
+    => Path Abs Dir -- ^ package directory
+    -> RIO env (P.PackageName, Path Abs File)
+findOrGenerateCabalFile pkgDir = do
+    hpack pkgDir
+    files <- filter (flip hasExtension "cabal" . toFilePath) . snd
+         <$> listDir pkgDir
+    -- If there are multiple files, ignore files that start with
+    -- ".". On unixlike environments these are hidden, and this
+    -- character is not valid in package names. The main goal is
+    -- to ignore emacs lock files - see
+    -- https://github.com/commercialhaskell/stack/issues/1897.
+    let isHidden ('.':_) = True
+        isHidden _ = False
+    case filter (not . isHidden . toFilePath . filename) files of
```

Comment:
How about if `toFilePath` gives you `./foobar.cabal`? This can happen in new-ish versions of `path` with relative files, I think.

Reply:
Another nit: it's better to use `fromRelFile` than `toFilePath` for future-proofing.

Reply:
With the recent push, we use `fromRelPath`.

Reply:
It can't happen since the `filename` call will strip off any leading directories.